### PR TITLE
[14.0][MIG] purchase_blanket_order

### PR DESCRIPTION
**Pull request by ChrisOForgeFlow**
Standard Migration
@ForgeFlow

### CODE REVIEW

**Review by LoisRForgeFlow — APPROVED** [review]
Code review LGTM

File: purchase_blanket_order/wizard/create_purchase_orders.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+        # in case the cron hasn't run
+        self.env["purchase.blanket.order"].expire_orders()
+        if not self.env.context.get("active_id"):
+            return False
+        blanket_order = self.env["purchase.blanket.order"].search(
+            [("id", "=", self.env.context["active_id"])], limit=1
+        )
+        if blanket_order.state == "expired":
+            raise UserError(
+                _("You can't create a purchase order from " "an expired blanket order!")
```
Comment (HviorForgeFlow) [inline purchase_blanket_order/wizard/create_purchase_orders.py]: Remove duplicated double-quotes

File: purchase_blanket_order/models/blanket_orders.py [hunk 2]
```diff
@@ -0,0 +1,614 @@
...
+
+    @api.onchange("product_id", "original_uom_qty")
+    def onchange_product(self):
+        precision = self.env["decimal.precision"].precision_get(
+            "Product Unit of Measure"
+        )
+        if self.product_id:
+            name = self.product_id.name
+            if not self.product_uom:
+                self.product_uom = self.product_id.uom_id.id
```
Comment (HviorForgeFlow) [inline purchase_blanket_order/models/blanket_orders.py]: Apply https://github.com/OCA/purchase-workflow/pull/1470

**Review by HviorForgeFlow — COMMENTED** [review]
Code Review, minor comments.

File: purchase_blanket_order/models/blanket_orders.py [hunk 3]
```diff
@@ -0,0 +1,614 @@
...
+        if len(purchase_orders) > 0:
+            action["domain"] = [("id", "in", purchase_orders.ids)]
+            action["context"] = [("id", "in", purchase_orders.ids)]
+        else:
+            action = {"type": "ir.actions.act_window_close"}
+        return action
+
+    def action_view_purchase_blanket_order_line(self):
+        action = self.env.ref(
+            "purchase_blanket_order" ".act_open_purchase_blanket_order_lines_view_tree"
```
Comment (HviorForgeFlow) [inline purchase_blanket_order/models/blanket_orders.py]: Same

File: purchase_blanket_order/wizard/create_purchase_orders.py [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+        for line in bo_lines:
+
+            if line.order_id.state != "open":
+                raise UserError(
+                    _("Purchase Blanket Order %s is not open") % line.order_id.name
+                )
+
+            line_company_id = line.company_id and line.company_id.id or False
+            if company_id is not False and line_company_id != company_id:
+                raise UserError(_("You have to select lines " "from the same company."))
```
Comment (HviorForgeFlow) [inline purchase_blanket_order/wizard/create_purchase_orders.py]: Same

**Review by HviorForgeFlow — APPROVED** [review]

**Pull request merged**